tracing-appender/src/non_blocking.rs

```diff
@@ -8,14 +8,21 @@ use std::sync::Arc;
 use std::thread::JoinHandle;
 use tracing_subscriber::fmt::MakeWriter;
 
+/// Default number of lines to buffer before dropping logs if `NonBlocking` is is lossy.
 pub const DEFAULT_BUFFERED_LINES_LIMIT: usize = 128_000;
 
+/// `WorkerGuard` is responsible for informing the writer to flush logs in the case of a panic.
+///
+/// This guard should be held for as long as possible and should not be dropped accidentally.
+/// It contains a reference to an `AtomicBool` which is used to inform the off-thread writer's
+/// worker to shutdown and flush logs.
 #[derive(Debug)]
 pub struct WorkerGuard {
```

Comment:
perhaps this should also have a `#[must_use]` attribute so that users will be warned if it is ignored?

Reply:
added